cmd/thanos/compact.go

```diff
@@ -458,6 +459,51 @@ func runCompact(
 		return cleanPartialMarked()
 	}
 
+	if conf.compactionProgressMetrics {
+		g.Add(func() error {
+			ps := compact.NewCompactionProgressCalculator(reg, tsdbPlanner)
+			var ds *compact.DownsampleProgressCalculator
+			if !conf.disableDownsampling {
+				ds = compact.NewDownsampleProgressCalculator(reg)
+			}
+
+			return runutil.Repeat(5*time.Minute, ctx.Done(), func() error {
+
+				if err := sy.SyncMetas(ctx); err != nil {
+					return errors.Wrapf(err, "could not sync metas")
+				}
+
+				metas := sy.Metas()
+				groups, err := grouper.Groups(metas)
+				if err != nil {
+					return errors.Wrapf(err, "could not group original metadata")
+				}
+
+				ps.CompactProgressMetrics.NumberOfCompactionRuns.Reset()
+				ps.CompactProgressMetrics.NumberOfCompactionBlocks.Reset()
```

Comment:
Can you put the reset method inside `ProgressCalculate`? IMO it is best to put it above https://github.com/thanos-io/thanos/pull/4801/files#diff-be21b0381ac14ee139355d057b390aa69ff338fe1c748ae8a475ce46f3583575R564.